stac_geoparquet/to_arrow.py

```diff
@@ -55,13 +56,18 @@ def _stac_items_to_arrow(items: Sequence[dict[str, Any]]) -> pa.Table:
     Returns:
         _description_
     """
-    # TODO:!! Can just call pa.array() on the list of python dicts!!
-    with NamedTemporaryFile("w+b", suffix=".json") as f:
-        for item in items:
-            f.write(json.dumps(item, separators=(",", ":")).encode("utf-8"))
-            f.write("\n".encode("utf-8"))
-
-        return _stac_ndjson_to_arrow(f)
+    # TODO: Handle STAC items with different schemas
+    # This will fail if any of the items is missing a field since the arrays
+    # will be different lengths.
+    d = defaultdict(list)
+
+    for item in items:
+        for k, v in item.items():
+            d[k].append(v)
+
+    arrays = {k: pa.array(v) for k, v in d.items()}
+    t = pa.table(arrays)
```

Comment:
I don't think you even have to do this...? You're manually constructing a dictionary of arrays, when you can pass the list of dicts directly.

```py
import pyarrow as pa
d1 = {
    'a': 1,
    'b': {
        'c': 'foo'
    }
}
d2 = {
    'a': 2,
    'b': {
        'c': 'bar'
    }
}
pa.array([d1, d2])
# <pyarrow.lib.StructArray object at 0x12665ba60>
# -- is_valid: all not null
# -- child 0 type: int64
#   [
#     1,
#     2
#   ]
# -- child 1 type: struct<c: string>
#   -- is_valid: all not null
#   -- child 0 type: string
#     [
#       "foo",
#       "bar"
#     ]
```
obviously this requires that the dicts have the same schema, but this is a requirement for now anyways

Reply:
> obviously this requires that the dicts have the same schema, but this is a requirement for now anyways

Actually, this handles schema resolution automatically, just like `pa.json.read_json` does. The only downside is that all the input data has to fit in memory at once for the schema resolution:

```py
import pyarrow as pa
d1 = {
    'a': 1,
    'b': {
        'c': 'foo'
    }
}
d2 = {
    'a': 2,
    'b': {
        'c': 'bar',
        'd': 'baz'
    }
}
pa.array([d1, d2])
# <pyarrow.lib.StructArray object at 0x105db5000>
# -- is_valid: all not null
# -- child 0 type: int64
#   [
#     1,
#     2
#   ]
# -- child 1 type: struct<c: string, d: string>
#   -- is_valid: all not null
#   -- child 0 type: string
#     [
#       "foo",
#       "bar"
#     ]
#   -- child 1 type: string
#     [
#       null,
#       "baz"
#     ]
```